extension/obs/src/index.ts

```diff
@@ -37,75 +58,80 @@ class OBS extends EventEmitter {
         setTimeout(() => this.connect(), 5000);
       });
 
-      this.conn.on('SwitchScenes', (data) => {
+      this.conn.on('CurrentProgramSceneChanged', (data) => {
         const lastScene = this.currentScene;
-        if (lastScene !== data['scene-name']) {
-          this.currentScene = data['scene-name'];
+        if (lastScene !== data.sceneName) {
+          this.currentScene = data.sceneName;
           this.emit('currentSceneChanged', this.currentScene, lastScene);
         }
       });
 
-      this.conn.on('ScenesChanged', async () => {
-        const scenes = await this.conn.send('GetSceneList');
-        this.sceneList = scenes.scenes.map((s) => s.name);
+      this.conn.on('SceneListChanged', async ({ scenes }) => {
+        this.sceneList = (scenes as { sceneIndex: number, sceneName: string }[])
+          .sort((s, b) => b.sceneIndex - s.sceneIndex)
+          .map((s) => s.sceneName);
         this.emit('sceneListChanged', this.sceneList);
       });
 
-      this.conn.on('StreamStarted', () => {
-        this.streaming = true;
+      this.conn.on('StreamStateChanged', ({ outputActive }) => {
+        this.streaming = outputActive;
         this.emit('streamingStatusChanged', this.streaming, !this.streaming);
       });
 
-      this.conn.on('StreamStopped', () => {
-        this.streaming = false;
-        this.emit('streamingStatusChanged', this.streaming, !this.streaming);
-      });
-
-      this.conn.on('error', (err) => {
+      this.conn.on('ConnectionError', (err) => {
         nodecg.log.warn('[OBS] Connection error');
         nodecg.log.debug('[OBS] Connection error:', err);
       });
+
+      this.conn.on('Identified', () => {
+        // wait a few seconds to make sure stuff is loaded.
```

Comment:
Do you know what "stuff" is in this context that is needing an artificial delay? I don't usually like them as it can cause race conditions but willing to accept if it's an easier solution.

Reply:
In this context "stuff" is the entirety of  obs, basically

Reply:
This is not specific to v5 of the websocket. It only started happening to me on OBS 30